removefiles/activity.go

```diff
@@ -0,0 +1,51 @@
+package removefiles
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+)
+
+const ActivityName = "remove-files-activity"
+
+type Activity struct {
+	logger logr.Logger
+}
+
+func NewActivity(logger logr.Logger) *Activity {
+	return &Activity{logger: logger}
+}
+
+type ActivityParams struct {
+	// RemoveNames is the comma separated list of filenames that should be
+	// removed from Path.
+	RemoveNames string
+
+	// Path is the directory from which files should be removed.
+	Path string
+}
+
+type ActivityResult struct {
+	// Count is the number of files removed from Path.
+	Count int
+}
+
+// Execute deletes any files named by RemoveNames from the given Path and
+// returns a count of deleted files.
+func (a *Activity) Execute(
+	ctx context.Context,
+	params *ActivityParams,
+) (*ActivityResult, error) {
+	a.logger.V(2).Info("Executing RemoveFilesActivity",
+		"RemoveNames", params.RemoveNames,
+		"Path", params.Path,
+	)
```

Comment:
Not sure if this adds a lot of value, considering it's a shared activity and that we only use the logger for this, while Temporal already logs:

```
[secure-zone-worker] 2024-04-15T17:59:56.794Z	V(1)	worker.temporal	log/with_logger.go:69	ExecuteActivity	{"level": "debug", "Namespace": "default", "TaskQueue": "secure-zone", "WorkerID": "1@secure-zone-api-worker-0@", "WorkflowType": "import-package", "WorkflowID": "import-package-25add6d3-8f5b-460c-90ac-ca836bd1e153", "RunID": "88dc3b6b-88c9-4831-b5da-aa938bb6054a", "Attempt": 1, "ActivityID": "35", "ActivityType": "extract-package"}
```

I'd not bother to share the logger just to log the activity call parameters.

Reply:
This is a common pattern we've used in a number of activities in SDPS Enduro:

- https://github.com/artefactual-sdps/enduro/blob/main/internal/workflow/activities/bundle.go#L57
- https://github.com/artefactual-sdps/enduro/blob/main/internal/workflow/activities/download.go#L45
- https://github.com/artefactual-sdps/enduro/blob/main/internal/workflow/activities/unarchive.go#L52

I've found it useful in debugging to see the parameters being passed to the activity, as well as confirmation that the actual activity is running.